packages/components/src/styles/components/stepper/nav.scss

```
@@ -0,0 +1,326 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// STEPPER > NAV
+//
+
+@use "sass:map";
+@use "../../mixins/focus-ring" as *;
+@use "./step-indicator.scss" as *;
+
+.hds-stepper-nav__list,
+.hds-stepper-nav__step {
+  margin: 0;
+  padding: 0;
+  list-style: none;
+}
+
+.hds-stepper-nav__list {
+  display: flex;
+}
+
+// STEPPER > NAV > STEP
+
+.hds-stepper-nav__step {
+  width: 100%;
+}
+
+.hds-stepper-nav__step__content {
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  padding: 0;
+  background-color: transparent;
+  border: none;
+}
+
+.hds-stepper-nav__step__progress {
+  position: absolute;
+  top: -14px;
+  left: 0;
+  z-index: 0;
+  display: flex;
+  justify-content: center;
+  width: 100%;
+
+  &::before,
+  &::after {
+    position: absolute;
+    top: 10px;
+    left: 0;
+    z-index: -1;
+    width: 100%;
+    height: 4px;
+    content: "";
+  }
+
+  &::before {
+    background-color: var(--token-color-palette-alpha-200);
+  }
+
+  &::after {
+    background-color: var(--token-color-palette-neutral-700);
+  }
+}
+
+.hds-stepper-nav .hds-stepper-nav__step:first-of-type .hds-stepper-nav__step__progress::before,
+.hds-stepper-nav .hds-stepper-nav__step:first-of-type .hds-stepper-nav__step__progress::after {
+  border-radius: 20px 0 0 20px;
+}
+
+.hds-stepper-nav .hds-stepper-nav__step:last-of-type .hds-stepper-nav__step__progress::before,
+.hds-stepper-nav .hds-stepper-nav__step:last-of-type .hds-stepper-nav__step__progress::after {
+  border-radius: 0 20px 20px 0;
+}
+
+.hds-stepper-nav__step__text {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  padding: 14px 12px 12px 12px;
+  border-radius: 0 0 var(--token-border-radius-small) var(--token-border-radius-small);
+}
+
+.hds-stepper-nav__step__title {
+  color: var(--token-color-foreground-primary);
+}
+
+.hds-stepper-nav__step__description {
+  margin-top: 4px;
+  color: var(--token-color-foreground-primary);
+}
+
+// STAUSES
+
+// Complete & Active
+
+.hds-stepper-nav__step--complete .hds-stepper-nav__step__title,
+.hds-stepper-nav__step--active .hds-stepper-nav__step__title {
+  color: var(--token-color-foreground-strong);
+}
+
+// Incomplete
+
+.hds-stepper-nav__step--incomplete .hds-stepper-nav__step__progress::after {
+  width: 0;
+}
+
+// Active
+
+.hds-stepper-nav__step--active {
+  .hds-stepper-nav__step__btn {
+    @include hds-focus-ring-with-pseudo-element(
+      $top: -3px,
+      $right: -1px,
+      $bottom: -1px,
+      $left: -1px,
+      $radius: 0 0 6px 6px
+    );
+  }
+
+  .hds-stepper-nav__step__progress::after {
+    width: calc(50% - 16px); // 1/2 width of icon (12px) + 4px margin
+    border-radius: 0 20px 20px 0;
```

Alignment elsewhere here too

```suggestion
    border-radius: 0 var(--token-border-radius-x-small) var(--token-border-radius-x-small) 0;
```